src/ssm.js

```diff
@@ -100,8 +103,14 @@
         fireAllMethodsInArray(leaveMethods);
         fireAllMethodsInArray(enterMethods);
         fireAllMethodsInArray(resizeMethods);
+
+        if (changed) {
```

Comment:
Thanks, I really appreciate your contribution and I think this is a great idea, just one last comment (and apologies I didnt notice this the first time) and of course correct me if I am wrong, it looks like the onStateChange will only run when states are entered but not when they have left, I would assume you would want to flag the state being changed whenever its changed, so on enter or on leave?